### ISSUE TITLE
istioctl failed to use --set with list

### ISSUE THREAD

**Issue by carolynhu** [istio/istio#20916]
I followed this command: https://preliminary.istio.io/docs/ops/configuration/telemetry/in-proxy-service-telemetry/

I'm using `istio-1.5-alpha.ff8fbc8cadad6fb9e3264c6e09ecb97badce7513` istioctl.
```
istioctl manifest apply --set values.telemetry.enabled=true,values.telemetry.v1.enabled=false,values.telemetry.v2.enabled=true,values.telemetry.v2.prometheus.enabled=true
```
to enable mixerless telemetry.

But it throws errors:

```
Error: failed to generate and apply manifests, error: failed to generate tree from the set overlay, error: bad argument values.telemetry.enabled=true,values.telemetry.v1.enabled=false,values.telemetry.v2.enabled=true,values.telemetry.v2.prometheus.enabled=true: expect format key=value
```

Well, when I apply those key=value pairs separately one by one, it works. There might be some chaining logic not correct there. Please take a look.

**Comment by howardjohn** [comment]
@ostromart did the change to allow lists break this maybe

**Comment by richardwxn** [comment]
I think this was fixed by a recent PR already, you can try newer build

**Comment by richardwxn** [comment]
ok seems like it is broken again for set with a list

**Comment by ostromart** [comment]
I'm actually more inclined to disallow this syntax and stick with the --set a=b --set c=d format. In that case it's a documentation update.

**Comment by esnible** [comment]
I broke this, and I broke it on purpose.  I didn't see this issue until today.

I introduced a chance in `--set` processing for https://github.com/istio/istio/issues/20381 .  The networking folks were doing `--set values.global.proxy.includeIPRanges="172.30.0.0/16,172.21.0.0/16"` and we had some doc pages that expected that to work.  There is no good way to support values that include commas **and** allow users to use commas as a short substitute for a second `--set`.

May I close this?
